corehq/ex-submodules/casexml/apps/case/xform.py

```diff
@@ -302,10 +345,13 @@ def _validate_indices(case):
                         case.get_id,
                         index.referenced_id,
                     )
+                else:
+                    if referenced_case.owner_id != case.owner_id:
+                        dirtiness_flags.append(DirtinessFlag(index.referenced_id, case.owner_id))
```

Comment:
good call. i just wrote the test wrong and then fixed it to match when it failed.